Solve Webkit E2E test flakiness

I've only seen it happen on Webkit

- [ ] maybe label string length matters? 
- [x] add snapshot on failing tests, so we have some more useful feedback to work with

https://github.com/maykinmedia/open-beheer/actions/runs/18523059453/job/52787358304?pr=294

```
=================================== FAILURES ===================================
___________________ test_create_informatieobjecttype[webkit] ___________________

page = <Page url='http://localhost:51149/OZ/'>
runner = <openbeheer.utils.gherkin_e2e.GherkinRunner object at 0x7f9e8dfe7890>

    @pytest.mark.e2e
    def test_create_informatieobjecttype(page: Page, runner: GherkinRunner):
        ServiceFactory.create(
            api_type=APITypes.ztc,
            api_root="http://localhost:8003/catalogi/api/v1",
            client_id="test-vcr",
            secret="test-vcr",
            slug="OZ",
        )
        helper = OpenZaakDataCreationHelper(ztc_service_slug="OZ")
        catalogus = helper.create_catalogus()
        assert catalogus.url
    
        UserFactory.create(username="johndoe", ***)
    
        runner.when.go_to_root_page(page)
        runner.when.user_logs_in(page, username="johndoe", ***)
    
        runner.then.path_should_be(page, "/OZ/")
    
>       runner.when.select_catalogus(page, catalogus)

src/openbeheer/informatieobjecttypen/tests/e2e/test_create_page.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/openbeheer/utils/gherkin_e2e.py:93: in select_catalogus
    option.click()
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/playwright/sync_api/_generated.py:15543: in click
    self._sync(
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/playwright/_impl/_locator.py:160: in click
    return await self._frame.click(self._selector, strict=True, **params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/playwright/_impl/_frame.py:549: in click
    await self._channel.send("click", self._timeout, locals_to_params(locals()))
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/playwright/_impl/_connection.py:69: in send
    return await self._connection.wrap_api_call(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <playwright._impl._connection.Connection object at 0x7f9e8ebf0e90>
cb = <function Channel.send.<locals>.<lambda> at 0x7f9e8c7f72e0>
is_internal = False, title = None

    async def wrap_api_call(
        self, cb: Callable[[], Any], is_internal: bool = False, title: str = None
    ) -> Any:
        if self._api_zone.get():
            return await cb()
        task = asyncio.current_task(self._loop)
        st: List[inspect.FrameInfo] = getattr(
            task, "__pw_stack__", None
        ) or inspect.stack(0)
    
        parsed_st = _extract_stack_trace_information_from_stack(st, is_internal, title)
        self._api_zone.set(parsed_st)
        try:
            return await cb()
        except Exception as error:
>           raise rewrite_error(error, f"{parsed_st['apiName']}: {error}") from None
E           playwright._impl._errors.TimeoutError: Locator.click: Timeout 30000ms exceeded.
E           Call log:
E             - waiting for get_by_text("Station class feeling hand participant according term instead. (HLVUF)")

/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/playwright/_impl/_connection.py:558: TimeoutError
=============================== warnings summary ===============================
../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148
../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148
../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148
../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148
../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148
../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148
  /opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/django/db/models/fields/__init__.py:1148: RemovedInDjango60Warning: The default scheme will be changed from 'http' to 'https' in Django 6.0. Pass the forms.URLField.assume_scheme argument to silence this warning, or set the FORMS_URLFIELD_ASSUME_HTTPS transitional setting to True to opt into using 'https' as the new default scheme.
    return form_class(**defaults)

../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/_pytest/nodes.py:329
  /opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/_pytest/nodes.py:329: PytestUnknownMarkWarning: Unknown pytest.mark.vcr - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    marker_ = getattr(MARK_GEN, marker)

../../../../../../opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/_pytest/nodes.py:329
  /opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/_pytest/nodes.py:329: PytestUnknownMarkWarning: Unknown pytest.mark.gh-128 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    marker_ = getattr(MARK_GEN, marker)

src/openbeheer/informatieobjecttypen/tests/e2e/test_create_page.py::test_create_informatieobjecttype[webkit]
  /opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/simple_certmanager/migrations/0003_signingrequest_and_more.py:84: RemovedInDjango60Warning: CheckConstraint.check is deprecated in favor of `.condition`.
    constraint=models.CheckConstraint(

src/openbeheer/informatieobjecttypen/tests/e2e/test_create_page.py::test_create_informatieobjecttype[webkit]
src/openbeheer/informatieobjecttypen/tests/e2e/test_list_page.py::test_retrieve_informatieobjecttypen[webkit]
  /opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/factory/django.py:182: DeprecationWarning: UserFactory._after_postgeneration will stop saving the instance after postgeneration hooks in the next major release.
  If the save call is extraneous, set skip_postgeneration_save=True in the UserFactory.Meta.
  To keep saving the instance, move the save call to your postgeneration hooks or override _after_postgeneration.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED src/openbeheer/informatieobjecttypen/tests/e2e/test_create_page.py::test_create_informatieobjecttype[webkit] - playwright._impl._errors.TimeoutError: Locator.click: Timeout 30000ms exceeded.
Call log:
  - waiting for get_by_text("Station class feeling hand participant according term instead. (HLVUF)")
========== 1 failed, 2 passed, 189 deselected, 11 warnings in 43.76s ===========
```

--- a/backend/src/openbeheer/utils/gherkin_e2e.py
+++ b/backend/src/openbeheer/utils/gherkin_e2e.py
@@ -175,6 +175,7 @@ class GherkinRunner:
             profile_button.click()
 
             logout_button = page.get_by_role("button", name="Logout")
+            logout_button.wait_for()
             logout_button.click()
 
         # Navigation

